### PR TITLE
HWKMETRICS-165

### PR DESCRIPTION
**Pull request by mwringe**
Add docker-push profile to push images out to a docker registry.

### CODE REVIEW

File: containers/pom.xml [hunk 1]
```diff
@@ -42,10 +42,32 @@
...
     <hawkular-metrics.image>${docker.user}/hawkular-metrics:${project.version}</hawkular-metrics.image>
     <cassandra.build>devel</cassandra.build>
-    <cassandra.image>${docker.user}/cassandra:${cassandra.version}-${cassandra.build}</cassandra.image>
+    <cassandra.image>${docker.user}/hawkular-cassandra:${cassandra.version}-${cassandra.build}</cassandra.image>
   </properties>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.jolokia</groupId>
```
Comment (tsegismont) [inline containers/pom.xml]: This should be in the pluginsManagement/plugins section

File: containers/hawkular-metrics-docker/pom.xml [hunk 2]
```diff
@@ -95,6 +95,27 @@
       </build>
     </profile>
     <profile>
+      <id>docker-push</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.jolokia</groupId>
+            <artifactId>docker-maven-plugin</artifactId>
+            <version>${docker-maven-plugin.version}</version>
```
Comment (tsegismont) [inline containers/hawkular-metrics-docker/pom.xml]: You don't need to declare the version if the plugin is declared in the pluginsManagement/plugins section

File: containers/hawkular-metrics-docker/pom.xml [hunk 3]
```diff
@@ -81,7 +81,7 @@
           <plugin>
             <groupId>org.jolokia</groupId>
             <artifactId>docker-maven-plugin</artifactId>
-            <version>0.11.2</version>
+            <version>${docker-maven-plugin.version}</version>
```
Comment (tsegismont) [inline containers/hawkular-metrics-docker/pom.xml]: Same here

File: containers/hawkular-metrics-docker/pom.xml [hunk 4]
```diff
@@ -55,14 +55,14 @@
       <plugin>
         <groupId>org.jolokia</groupId>
         <artifactId>docker-maven-plugin</artifactId>
-        <version>0.12.0</version>
+        <version>${docker-maven-plugin.version}</version>
```
Comment (tsegismont) [inline containers/hawkular-metrics-docker/pom.xml]: Same here

File: containers/pom.xml [hunk 5]
```diff
@@ -42,10 +42,32 @@
...
+        <artifactId>docker-maven-plugin</artifactId>
+        <version>0.12.0</version>
+      </dependency>
+      <dependency>
+        <groupId>io.fabric8</groupId>
```
Comment (tsegismont) [inline containers/pom.xml]: This should be in the pluginsManagement/plugins section

File: containers/cassandra-docker/pom.xml [hunk 6]
```diff
@@ -90,6 +90,27 @@
       </build>
     </profile>
     <profile>
+      <id>docker-push</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.jolokia</groupId>
+            <artifactId>docker-maven-plugin</artifactId>
+            <version>${docker-maven-plugin.version}</version>
```
Comment (tsegismont) [inline containers/cassandra-docker/pom.xml]: Same here

File: containers/cassandra-docker/pom.xml [hunk 7]
```diff
@@ -76,7 +76,7 @@
           <plugin>
             <groupId>org.jolokia</groupId>
             <artifactId>docker-maven-plugin</artifactId>
-            <version>0.11.2</version>
+            <version>${docker-maven-plugin.version}</version>
```
Comment (tsegismont) [inline containers/cassandra-docker/pom.xml]: Same here

File: containers/cassandra-docker/pom.xml [hunk 8]
```diff
@@ -50,14 +50,14 @@
       <plugin>
         <groupId>org.jolokia</groupId>
         <artifactId>docker-maven-plugin</artifactId>
-        <version>0.12.0</version>
+        <version>${docker-maven-plugin.version}</version>
```
Comment (tsegismont) [inline containers/cassandra-docker/pom.xml]: Same here

File: containers/Readme.md [hunk 9]
```diff
@@ -79,3 +79,15 @@ export KUBERNETES_CLIENT_CERTIFICATE_FILE=$OPENSHIFT_HOME/openshift.local.config
 export KUBERNETES_CLIENT_KEY_FILE=$OPENSHIFT_HOME/openshift.local.config/master/admin.key
 mvn clean install -Pcontainer-tests
 
+Deploying the Docker Images to a Registry
```
Comment (tsegismont) [inline containers/Readme.md]: I've been able to run the container tests successfully this morning. So I've updated the doc. Also I've converted it to Asciidoc which is the standard on all H* projects. After you've fixed the pluginMgmt section issues I'll merge this PR manually and will convert this last paragraph to Asciidoc.

**Comment by mwringe** [comment]
Updated to include requested fixes, updated to latest in master and squashed the commits together. Let me know if there are any more changes required.

**Comment by tsegismont** [comment]
Looks good to me. Thanks for the update @mwringe . I'll merge as soon as Travis is happy (it failed due to an OOM error, I've just restarted the build)

**Pull request merged**